lib/main.coffee

```diff
@@ -189,6 +191,13 @@ module.exports = Hydrogen =
             @kernelManager.restartRunningKernelFor grammar, (kernel) =>
                 @onKernelChanged kernel
 
+        else if command is 'shutdown-kernel'
+            @clearResultBubbles()
+            # Note that destroy alone does not shut down a WSKernel
+            kernel.shutdown()
+            @kernelManager.destroyRunningKernelFor grammar
+            @onKernelChanged()
```

Comment:
Note that `@kernel` is set to `null` at startup, but is set to `undefined` by `onEditorChanged` whenever the current tab doesn't have a running kernel (see `KernelManager::getRunningKernelFor`).